ballista/client/src/context.rs

```diff
@@ -137,13 +139,17 @@ impl BallistaContext {
     pub async fn standalone(
         config: &BallistaConfig,
         concurrent_tasks: usize,
+        table_factories: HashMap<String, Arc<dyn TableProviderFactory>>,
```

Comment:
Should we use `TableProviderSessionBuilder` here? And for the standalone case, can the table_factories be determined by the ballista config? Maybe we don't need this additional parameter for this interface.

Reply:
> can the table_factories be determined by the ballista config?

Unfortunately, I don't think so:

```
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct BallistaConfig {
```

I don't think it is logical for `TableProviderFactories` to implement `PartialEq` or `Eq`, and without making another DataFusion PR, they don't impl `Debug` at present either. Everything else in the `BallistaConfig` is a key-value pair, and extending it to include stateful runtime objects seems like a questionable choice.

I can however swap out the Map for an `Arc<dyn SessionBuilder>`, but then `client` needs to have a new dependency on `core`.

Reply:
I made it `Option`al with a default. Hopefully passing `None` will be less burdensome.

Reply:
Another option I just thought of: create a new struct (i.e. `BallistaSettings`) which wraps `BallistaConfig` but also adds `concurrent_tasks` and `table_factories` fields.